dev/src/bulk-writer.ts

```diff
@@ -743,6 +784,24 @@ export class BulkWriter {
     });
   }
 
+  /**
+   * Adds a 30% jitter to the provided backoff.
+   *
+   * @private
+   */
+  private static applyJitter(backoffMs: number): number {
+    if (backoffMs === 0) return 0;
+    // Random value in [-0.3, 0.3].
+    const jitter =
+      Math.random() *
+      DEFAULT_JITTER_FACTOR *
+      (Math.round(Math.random()) ? 1 : -1);
```

Comment:
Math.random() is not cheap. You can remove one call: Math.random() * 2 * DEFAULT_JITTER_FACTOR - DEFAULT_JITTER_FACTOR.

Reply:
Ooh that's some nice mathing! Refactored it slightly to make it more understandable (at least to me).